### ISSUE TITLE
Materialize the AlterMigrationRuleStatement

### ISSUE THREAD

**Issue by Pace2Car** [apache/shardingsphere#25261]
## Feature Request

**For English only**, other languages will not be accepted.

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot make decision by current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Describe the feature you would like.
`AlterInventoryIncrementalRuleStatement` is more like an abstract pipeline job statement, when using DistSQL `ALTER MIGRATION RULE`, the visitor will visit this statement.
I am considering changing AlterInventoryIncrementalRuleStatement to an abstract class, retaining the ability to expand, and replacing its previous work with AlterMigrationRuleStatement

After this chage, the relation is:
<img width="570" alt="image" src="https://user-images.githubusercontent.com/37339007/233586974-4e3a50f9-c87a-4f63-a92d-997d944b3973.png">

**Comment by Pace2Car** [comment]
I will work on this.

**Comment by boyjoy1127** [comment]
Will this issue be completed before June 15 which is the final date of version 5.4.0?

**Comment by Pace2Car** [comment]
No, this issue is not urgent at the moment, it will be completed in the follow-up plan.

**Comment by terrymanu** [comment]
The codebase has already been refactored to use the generic AlterTransmissionRuleStatement with jobType=MIGRATION to handle ALTER MIGRATION RULE, and the previous AlterInventoryIncrementalRuleStatement no longer exists. The original mismatch mentioned here has effectively been addressed by the current implementation, and no further action is needed. This issue can be closed.
